doc/release-notes/iceoryx2-unreleased.md

```diff
@@ -106,7 +106,51 @@
 
 ### API Breaking Changes
 
-1. Renamed `PublisherLoanError` into `LoanError`
+1. Add requirement that every payload and user header type must implement
+   `ZeroCopySend` for type safe shared memory usage
+   [#602](https://github.com/eclipse-iceoryx/iceoryx2/issues/602)
+
+   ```rust
+   // old
+   #[repr(C)]
+   pub struct TransmissionData {
+      // ...
+   }
+
+   #[repr(C)]
+   pub struct CustomHeader {
+      // ...
+   }
+
+   let service = node
+       .service_builder(&"ServiceName".try_into()?)
+       .publish_subscribe::<TransmissionData>()
+       .user_header::<CustomHeader>()
+       .open_or_create()?;
+
+   // new
+   use iceoryx2_bb_derive_macros::ZeroCopySend;
```

Comment:
Since this is a basic requirement for any iceoryx2 usage we should add this to `iceoryx2/src/prelude.rs`. Then the use directive does not need to be updated.

When this is done, this line can be removed.